src/net.rs

```diff
@@ -236,6 +240,9 @@ impl NetworkStream for HttpStream {
             Https(ref mut inner) => inner.get_mut().peer_name()
         }
     }
+
+    fn mark_dead(&mut self) {
```

Comment:
(At least, so far; it's possible at some future date when client keep-alives are built into hyper in some fashion, that this method will be interesting again... but I'm treating that as out of scope of this change.)